### PR TITLE
Add URL rewriting check

### PR DESCRIPTION
**Pull request by bennothommo**
This does a URL rewrite check, similar to the one suggested by @digininja here: https://github.com/octobercms/install/issues/104 (hat tip for his suggestion).

### CODE REVIEW

**Comment by digininja** [comment]
Looks good to me. You could check the response code when doing the test, from what I saw, a 500 means .htaccess is broken, probably meaning rewrite rules aren't allowed in it, and a 404 means the rewrite has failed as the page is missing.

**Comment by bennothommo** [comment]
@digininja I did see that code in your example, but I felt that the reason it failed didn't matter too much for the Installer - if it fails, it fails, and the user would then need to be directed to the docs to look at how to set up rewriting.

**Comment by digininja** [comment]
I just thought it would be nice to be able to say "Rewriting is broken,
probably because of ..."

**Comment by bennothommo** [comment]
@digininja I'll definitely consider implementing a reason down the track. I feel there can be some improvement with the reporting of all checks on that System Compatibility screen.

File: install_files/php/Installer.php [hunk 1]
```diff
@@ -111,6 +111,9 @@ protected function onCheckRequirement()
             case 'allowUrlFopenConfig':
                 $result = ini_get('allow_url_fopen');
                 break;
+            case 'urlRewriting':
+                $content = @file_get_contents($this->getBaseUrl() . 'rewrite/test/');
+                $result = ($content === 'Rewrite successful');
```
Comment (LukeTowers) [inline install_files/php/Installer.php:116]: Shouldn't this have a break statement here too?

**Comment by digininja** [comment]
It's really hard to tell from the phone UI, but it looks like the last
entry so it doesn't need one, but putting one in would probably be best in
case someone adds something after it and doesn't notice.

File: install_files/php/Installer.php (hunk 1 above)
Comment (bennothommo) [reply install_files/php/Installer.php:116]: @LukeTowers for code cleanliness, yes, but as its technically the last case, it doesn't *require* it.

Comment (LukeTowers) [reply install_files/php/Installer.php:116]: Of course :) I'm all about that code cleanliness though, same principle as trailing commas

**Pull request closed without merging**